bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/EnergyManagerHandler.java

```diff
@@ -0,0 +1,408 @@
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EnergyManagerHandler} is responsible for handling energy manager thing itself
+ * and handle data retrieval for the child things.
+ *
+ * @author Sven Carstens - Initial contribution
+ */
+@NonNullByDefault
+public class EnergyManagerHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EnergyManagerHandler.class);
+
+    private final EnergyManagerConnector connector;
+    private final SolarwattChannelTypeProvider channelTypeProvider;
+
+    private @Nullable ExpiringCache<Map<String, Device>> devicesCache;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private @Nullable ZoneId zoneId;
+
+    /**
+     * Guid of this energy manager itself.
+     */
+    private @Nullable String energyManagerGuid;
+
+    /**
+     * Runner for the {@link ExpiringCache} refresh.
+     *
+     * Triggers update of all child things.
+     */
+    private final Runnable refreshRunnable = () -> {
+        EnergyManagerHandler.this.updateChannels();
+        EnergyManagerHandler.this.updateAllChildThings();
+    };
+
+    /**
+     * Create the handler.
+     *
+     * @param thing for which the handler is responsible
+     * @param channelTypeProvider provider for the channels
+     * @param httpClient connect to energy manager via this client
+     */
+    public EnergyManagerHandler(final Bridge thing, final SolarwattChannelTypeProvider channelTypeProvider,
+            final HttpClient httpClient) {
+        super(thing);
+        this.logger.debug("{} created", this);
+        this.connector = new EnergyManagerConnector(httpClient);
+        this.channelTypeProvider = channelTypeProvider;
+    }
+
+    /**
+     * Get services which are provided by this handler.
+     *
+     * Only service discovery is provided by
+     * 
+     * @return collection containing our discovery service
+     */
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(SolarwattDevicesDiscoveryService.class);
+    }
+
+    /**
+     * Execute the desired commands.
+     *
+     * Only refresh is supported and relayed to all childs of this thing.
+     *
+     * @param channelUID for which the command is issued
+     * @param command command issued
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.logger.debug("handleCommand called for channel: {}, command: {}", channelUID, command);
+
+        if (command instanceof RefreshType) {
+            this.updateChannels();
+        }
+    }
+
+    /**
+     * Dynnamically updates all known channel states of the energy manager.
+     */
+    public void updateChannels() {
+        this.logger.trace("{} updateChannels", this);
```

Comment:
Your using the 'this' keyword a lot when I'm pretty certain you don't need to. You do not have two loggers, so it is not necessary to use it in this case. It would only be needed if you had a global logger defined, and also had one with the exact same name in this local function/method. The same comment probably applies to all your uses of the this keyword most likely.

Also as a second comment on this line, it looks like your using this instead of the debugger, Does the line in the log give any useful info other than the function has been run? If no then this line should be removed and the debugger used instead.

Every time the function runs it will use CPU up to check if the logger has trace enabled, so it will reduce cpu load to remove this line even if your logger is not set to trace.

Reply:
I really prefer the this. prefix. I often change programming languages and the minor color differences between local and instance names is not something I could ever remember.

Excessive logging is removed.